### DISCUSSION TITLE
tasmota-sensor.bin wrong sensor identification

### DISCUSSION THREAD

**Discussion by tgonzalez1969** [arendst/Tasmota#10276] [Q&A] [answered]
Hi,

Let me describe my issue.

I've trying to control a VL53L0x Sensor -CJVL53L0XV2- (distance time-of-flight) with a tasmotized ESP01s. I use tasmota-sensors.bin file and the configuration is i2C SDA in GPIO0 and I2C SLC in GPIO2.

Tasmota identifies it quickly and without problems, but it says it is a TSL2561 luminance sensor ¡¡¿?!! giving the following MQTT message:
4:06:01 MQT: tele/sensor/gasoleo/SENSOR = {"Time":"2020-12-27T14:06:01","TSL2561":{"Illuminance":0.000,"IR":0,"Broadband":0}}

I've purchased a new sensor to check if there is a HW problem. I'm waiting for its delivery in order to make some tests but meanwhile I ask here because i'm not sure if i'm doing something wrong, there is something I've lost or any previous configuration i must implement.

Thanks in advance.

**Comment by sfromis** [comment, marked as answer]
Sometimes a sensor driver (like the TSL2561 one) may react to a device it should not. You can disable the offending driver to avoid that, in this case with the command `I2CDriver16 0`. In principle, it is a good plan to disable all unused drivers, but most drivers do not cause such troubles. Driver numbers (and their I2C addresses) are in the docs page about [I2CDevices](https://tasmota.github.io/docs/I2CDevices/).

**Reply by tgonzalez1969** [reply]
I've disabled i2cdriver16 (with the command you said) but now no "tele/.../SENSOR" message is sent. I suppose that as I disabled TSL2561 driver, now I have to enable VL53L0x driver. I've tried "i2cdriver31 1" without success. I get (when command is sent) 9:31:39 MQT: stat/sensor/gasoleo/RESULT = {"I2CDriver":"7,8,9,10,11,12,13,14,15,16,17,18,20,24,29,36,41,42,44,46,48"} as if only these drivers are available. I can read in the page you linked that "they mostly need to be enabled at compile time to solve possible address conflicts". is this the problem? Sorry for my inexperience, but it seems the solution is some levels above me, so I thank your patience.

**Reply by sfromis** [reply]
Yes, `USE_VL53L0X` is not included in any standard [builds](https://github.com/arendst/Tasmota/blob/development/BUILDS.md). You need to [compile](https://tasmota.github.io/docs/Compile-your-build/) your own binary, not a hard thing to do.

**Reply by tgonzalez1969** [reply]
Thanks, I will try. I'll come back with the result

**Reply by tgonzalez1969** [reply]
Finally, as sugested, I've compiled the tasmota binary. 

In the documentation it is sugested to use _user_config_override.h_ to include the customization, but after a couple of non successful atempts, I've changed the _my_user_config.h_ (not recomended). I've only deleted the "//"comment marks before "#define USE_VL53L0X"
And compiled. 

To compile the binaries i've used Gitpod. It automaticaly loads tasmota project in my workpalce, so I don't have to worry aboot lost librarys or other stuff.

Thanks Sfromis for your help
